Source/Core/DolphinNoGUI/CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+if(APPLE)
+	set(DOLPHIN_EXE_BASE Dolphin)
+else()
+	set(DOLPHIN_EXE_BASE dolphin-emu)
+endif()
+
+set(NOGUI_SRCS MainNoGUI.cpp)
+
+list(APPEND LIBS core uicommon cpp-optparse)
+
+if(USE_X11 OR ENABLE_HEADLESS)
+	set(DOLPHIN_NOGUI_EXE ${DOLPHIN_EXE_BASE}-nogui)
+	add_executable(${DOLPHIN_NOGUI_EXE} ${NOGUI_SRCS})
```

Comment:
No, what people meant was to get a different output name, and they achieved that by changing the target name.
Targets are logical entities that are always the same and they should have a constant name. Output name is fine to change.